## Final Project Submission

Please fill out:
* Student name: 
* Student pace: self paced / part time / full time
* Scheduled project review date/time: 
* Instructor name: 
* Blog post URL:


# 1.Introduction

# 1.1 Overview

The giant tech company Microsoft has decided that its time to venture into creating original video content and
therefore want to creat a movie studio
For this project, I will use exploratory data analysis to generate insights to  guide them on the way forward


# 1.2 Business understanding

Microsoft sees all the big companies creating original video content and they want to get in on the fun. 
They have decided to create a new movie studio, but they don’t know anything about creating movies. 
I have  been charged with exploring what types of films are currently doing the best at the box office. 
and give three recommendation

For me to translate these findings into actionable insights in order to help 
<br>Microsoft build a  new movie studio I should ask the important business questions <br>
<br>     1.Which type of films are the highest gross?<br>
<br>      2.What are the highest rated movies?<br>
 <br>    3.What genres are the highest grosssing and highest rated movies?<br>

Completing data analysis and answering the above questions will solve Microsost's quest for the Movie industry. 
They will have more information on the type of movies they are suppose to air in there anticipated Movie industry

# 2.  Data understanding

The data I am going to use in this Data analysis is  from Various websites contain information about the Box office
These datasets will enable me to understand the trends in the movie industry and therfore I will be able to produce the relevant 
information for microsoft so that they will be able to correctly excecute they dream of becoming succesful in the movie industry

These  datasets include<br>
 **movie_gross.csv**<br>
 **movie_info.tsv**<br>
 **rt.reviews.tsv**<br>
 **tmdb.movies.csv**<br>
 **movie_budgets.csv**<br>

are in the zippedData folder are from the following websites<br>
[Box Office Mojo](https://www.boxofficemojo.com/)<br>
[IMDB](https://www.imdb.com/)<br>
[Rotten Tomatoes](https://www.rottentomatoes.com/)<br>
[TheMovieDB](https://www.themoviedb.org/)<br>
[The Numbers](https://www.the-numbers.com/)<br>

# 2.1 Loading Libraries and Datasets

In [1967]:
    # importing the packages I will be using for this project
    import zippedData
    import numpy as np
    import sqlite3
    import pandas as pd
    conn = sqlite3.connect("data/im.db")
    import csv
    import matplotlib.pyplot as plt
    %matplotlib inline
    import seaborn as sns
    from collections import Counter

In [1968]:
#loading datasets
df_movie_gross = pd.read_csv('data/bom.movie_gross.csv')
df_movie_info = pd.read_table('data/rt.movie_info.tsv')
df_reviews = pd.read_table('data/rt.reviews.tsv',encoding = 'unicode_escape' )
df_movies= pd.read_csv('data/tmdb.movies.csv')
df_movie_budgets = pd.read_csv('data/tn.movie_budgets.csv')



# 2.2 Previewing the Datasets and getting information about dataset

# 2.2.1 df_movie_gross

In [1969]:
#previewing df_movie_gross

df_movie_gross.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [1970]:
# Accessing more information about df_movie_gross
df_movie_gross.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


This shows that the datas has 3 data types : object, float64 and interger.
  The column names include title, studio, domestic_gross, foreign_gross and year
  

In [1971]:
#checking the shape

df_movie_gross.shape

(3387, 5)

This shows that the data 3387 rows and 5 columns

In [1972]:
#checking for descriptive statistics

df_movie_gross.describe

<bound method NDFrame.describe of                                             title      studio  domestic_gross  \
0                                     Toy Story 3          BV     415000000.0   
1                      Alice in Wonderland (2010)          BV     334200000.0   
2     Harry Potter and the Deathly Hallows Part 1          WB     296000000.0   
3                                       Inception          WB     292600000.0   
4                             Shrek Forever After        P/DW     238700000.0   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.          6200.0   
3383                  Edward II (2018 re-release)          FM          4800.0   
3384                                     El Pacto        Sony          2500.0   
3385                                     The Swan  Synergetic          2400.0   
3386                            An Actor Prepares       Grav.          1700

Above are the descriptive statistics of the data

# 2.2.2 Previewing df_movie_info

In [1973]:
#previewing df_movie_info

df_movie_info.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [1974]:
#checking the data information

df_movie_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


This shows that the datas has 2 data types : object and integer.
  The columns include id, synopsis, rating, director, writer, theater_date, dvd_date, currency, box_office, runtime and studio

In [1975]:
#cheking the shape of data

df_movie_info.shape

(1560, 12)

The data has 1560 columns and 12 rows

In [1976]:
#checking for descriptive statistics

df_movie_info.describe()

id
count  1560.000000
mean   1007.303846
std     579.164527
min       1.000000
25%     504.750000
50%    1007.500000
75%    1503.250000
max    2000.000000

Above are the descriptive statistics of the data

# 2.2.3 Previewing df_reviews


In [1977]:
#previewing df_reviews 

df_reviews.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [1978]:
#checking the data information

df_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


The data has 2 data types : Interger and object. 
The columns include id, review, rating, fresh, critic, top_critic, publisher and date

In [1979]:
#checking the shape of the dataset 

df_reviews.shape

(54432, 8)

The data has 54432 columns and 8 rows

In [1980]:
#checking the dataset information

df_reviews.describe()

id    top_critic
count  54432.000000  54432.000000
mean    1045.706882      0.240594
std      586.657046      0.427448
min        3.000000      0.000000
25%      542.000000      0.000000
50%     1083.000000      0.000000
75%     1541.000000      0.000000
max     2000.000000      1.000000

Above are the descriptive statistics of the data

# 2.2.4 Previewing df_movies

In [1981]:
#previewing the df_movies

df_movies.head()

Unnamed: 0            genre_ids     id original_language  \
0           0      [12, 14, 10751]  12444                en   
1           1  [14, 12, 16, 10751]  10191                en   
2           2        [12, 28, 878]  10138                en   
3           3      [16, 35, 10751]    862                en   
4           4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [1982]:
#checking the dataset information

df_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         26517 non-null  int64  
 1   genre_ids          26517 non-null  object 
 2   id                 26517 non-null  int64  
 3   original_language  26517 non-null  object 
 4   original_title     26517 non-null  object 
 5   popularity         26517 non-null  float64
 6   release_date       26517 non-null  object 
 7   title              26517 non-null  object 
 8   vote_average       26517 non-null  float64
 9   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(3), object(5)
memory usage: 2.0+ MB


The data has 3 types : interger, object and float. 
The columns include genre_ids, id, original_langauge, oriinal_title, popularity, release_date, title, vote_avarage and vote_count

In [1983]:
#checking the shape of the dataset

df_movies.shape

(26517, 10)

The data has 26517 rows and 10 columns 

In [1984]:
#checking the descriptive statistics

df_movies.describe()

Unnamed: 0             id    popularity  vote_average    vote_count
count  26517.00000   26517.000000  26517.000000  26517.000000  26517.000000
mean   13258.00000  295050.153260      3.130912      5.991281    194.224837
std     7654.94288  153661.615648      4.355229      1.852946    960.961095
min        0.00000      27.000000      0.600000      0.000000      1.000000
25%     6629.00000  157851.000000      0.600000      5.000000      2.000000
50%    13258.00000  309581.000000      1.374000      6.000000      5.000000
75%    19887.00000  419542.000000      3.694000      7.000000     28.000000
max    26516.00000  608444.000000     80.773000     10.000000  22186.000000

Above are the descriptive statistics of the data

# 2.2.5 Previewing df_movie_budgets


In [1985]:
#previewing the df_movie_budgets

df_movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [1986]:
#checking the dataset information

df_movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


The data has 2 types : interger and object.
The columns includes id, rerlease_date, production_budget, domestic_gross and worldwide_gross

In [1987]:
#checking the shape of the dataset

df_movie_budgets.shape

(5782, 6)

The data has 5782 columns nad 6 rows

In [1988]:
#checking for descriptive statistics

df_movie_budgets.describe()

id
count  5782.000000
mean     50.372363
std      28.821076
min       1.000000
25%      25.000000
50%      50.000000
75%      75.000000
max     100.000000

Above are the descriptive statistics of the data

# 2.2.6 Previewing df_movie_basics

In [1989]:
#previewing df_movie_basics

df_movie_basics = pd.read_sql_query("""
SELECT *
FROM movie_basics
;
""", conn)
df_movie_basics.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [1990]:
#checking the dataset information

df_movie_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


The data has 3 data types : object, interger and float.
    Columns include movie_id, primary_title, original_title, start_year, runtime_minutes and genres

In [1991]:
#checking the shape of the dataset

df_movie_basics.shape

(146144, 6)

The data has 146144 columns and 6 rows

In [1992]:
#checking for descriptive statistics

df_movie_basics.describe()

start_year  runtime_minutes
count  146144.000000    114405.000000
mean     2014.621798        86.187247
std         2.733583       166.360590
min      2010.000000         1.000000
25%      2012.000000        70.000000
50%      2015.000000        87.000000
75%      2017.000000        99.000000
max      2115.000000     51420.000000

Above are the descriptive statistics of the data

# 2.2.7 Previewing df_movie_ratings

In [1993]:
#previewing movie_ratings

df_movie_ratings = pd.read_sql_query("""
SELECT *
FROM movie_ratings
;
""", conn)
df_movie_ratings.head()

movie_id  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [1994]:
#checking the dataset information

df_movie_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


The data has 3 data types : object, interger and float. 
    The columns are movie_id, averagerating and numvotes

In [1995]:
#checking the shape of the dataset

df_movie_basics.shape

(146144, 6)

The data has 73856 columns and 3 rows

In [1996]:
#checking for descriptive statistics

df_movie_basics.describe()

start_year  runtime_minutes
count  146144.000000    114405.000000
mean     2014.621798        86.187247
std         2.733583       166.360590
min      2010.000000         1.000000
25%      2012.000000        70.000000
50%      2015.000000        87.000000
75%      2017.000000        99.000000
max      2115.000000     51420.000000

Above are the descriptive statistics of the data

# 2.2.8 Relevant dataset and the limitation

After carefully scrutinizing the above datasets, 3 datasets stand out and they most relevant for my analysis and are able to answer the 3 business questions
This are 3 datasets<br> **1. df_movie_gross** <br> **2. df_movie_basics**<br> **3. df_movie_ratings**

One  limitation is the movie industry evolves drastically and the data we are using may cannot anticipate the changes that
may occur. A perfect example is that as the years progress, the new generation may be inclined to new types of genres of movies.
Hence by the time Microsoft has already started there movie industry, they could be changes is movie preference

# 2.3 Data Preparation and Data cleaning

# 2.3.1 df_movie_gross

In [1997]:
#previewing df_movie_basics

df_movie_gross.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [1998]:
#getting shape
df_movie_gross.shape

(3387, 5)

In [1999]:
print("Number of rows",df_movie_gross.shape[0])
print("Number of columns",df_movie_gross.shape[1])

Number of rows 3387
Number of columns 5


In [2000]:
# Accessing more information about df_movie_gross

df_movie_gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


All column data types above correspond to the data inside them

In [2001]:
#checking for null values

df_movie_gross.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

The dataset has null values

In [2002]:
# calaculating percentage of missing data

percent_missing = df_movie_gross.isnull().sum() * 100 / len(df_movie_gross)
percent_missing

title              0.000000
studio             0.147623
domestic_gross     0.826690
foreign_gross     39.858282
year               0.000000
dtype: float64

Now that we know how many the percentage of  missing values in each column, From the above data, 
Since grossing is a factor that is to be investigated, retaining these columns would distort the data,
hence, these rows need to be dropped.

In [2003]:
#dropping null values

df_movie_gross.dropna(inplace=True)
df_movie_gross

title        studio  \
0                                           Toy Story 3            BV   
1                            Alice in Wonderland (2010)            BV   
2           Harry Potter and the Deathly Hallows Part 1            WB   
3                                             Inception            WB   
4                                   Shrek Forever After          P/DW   
...                                                 ...           ...   
3275                                    I Still See You           LGF   
3286                              The Catcher Was a Spy           IFC   
3309                                         Time Freak    Grindstone   
3342  Reign of Judges: Title of Liberty - Concept Short  Darin Southa   
3353            Antonio Lopez 1970: Sex Fashion & Disco            FM   

      domestic_gross foreign_gross  year  
0        415000000.0     652000000  2010  
1        334200000.0     691300000  2010  
2        296000000.0     664300000  2010  
3        292600000.0     535700000  2010  
4        238700000.0     513900000  2010  
...              ...           ...   ...  
3275          1400.0       1500000  2018  
3286        725000.0        229000  2018  
3309         10000.0        256000  2018  
3342         93200.0          5200  2018  
3353         43200.0         30000  2018  

[2007 rows x 5 columns]

In [2004]:
#checkin the shape of data

df_movie_gross.shape

(2007, 5)

We now have 2007 rows and 5 columns

In [2005]:
# check for duplicates in the data

df_movie_gross.duplicated(subset=['title']).sum()

0

The data has no duplicates

In [2006]:
#checking information of the refined dataset

df_movie_gross.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2007 entries, 0 to 3353
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           2007 non-null   object 
 1   studio          2007 non-null   object 
 2   domestic_gross  2007 non-null   float64
 3   foreign_gross   2007 non-null   object 
 4   year            2007 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 94.1+ KB


It can be seen that the domestic_gross column is of type float. It will be preferable to convert in into interger type
because that column will be usefull in answering the first business question

In [2007]:
# converting the columns to integers

df_movie_gross['domestic_gross'] = df_movie_gross['domestic_gross'].astype('int')

In [2008]:
#checking if there there is changes in the columns

df_movie_gross.head(3)

title studio  domestic_gross  \
0                                  Toy Story 3     BV       415000000   
1                   Alice in Wonderland (2010)     BV       334200000   
2  Harry Potter and the Deathly Hallows Part 1     WB       296000000   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010

In [2009]:
#checking for null values in the refined datset

df_movie_gross.isna().sum()

title             0
studio            0
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

As we can see from the above the all the null values have been drop

In [2010]:
# check for duplicates in the data

df_movie_gross.duplicated(subset=['title']).sum()

0

The dataset doesnt have duplicates

In [2011]:
#previewing the dataset

df_movie_gross.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV       415000000   
1                   Alice in Wonderland (2010)     BV       334200000   
2  Harry Potter and the Deathly Hallows Part 1     WB       296000000   
3                                    Inception     WB       292600000   
4                          Shrek Forever After   P/DW       238700000   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

# 2.3.2 df_movie_basics

In [2012]:
#preview of the dataset

df_movie_basics.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [2013]:
# Accessing more information about df_movie_basics

df_movie_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


The dataset appears to have 3 data types : float interger and object and they conforms to the data inside

In [2014]:
#getting the shape of the dataset

df_movie_basics.shape

(146144, 6)

In [2015]:
print("Number of rows",df_movie_basics.shape[0])
print("Number of columns",df_movie_basics.shape[1])

Number of rows 146144
Number of columns 6


In [2016]:
# Accessing more information about df_movie_basics

df_movie_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


All column data types above correspond to the data inside them

In [2017]:
#checking for null values

df_movie_basics.isna().sum()

movie_id               0
primary_title          0
original_title        21
start_year             0
runtime_minutes    31739
genres              5408
dtype: int64

The dataset has null values in all columns except movie_id, primary_title
 and start_year. There several null values especially in the columns of runtime_minutes and genres. 

In [2018]:
# calculating percentage of missing data

percent_missing = df_movie_basics.isnull().sum() * 100 / len(df_movie_basics)
percent_missing

movie_id            0.000000
primary_title       0.000000
original_title      0.014369
start_year          0.000000
runtime_minutes    21.717621
genres              3.700460
dtype: float64

    The runtime_minutes column is not important in answering the business questions  formulated but
    genres column does. But it is appropriate that to drop the runtime_minutes  column and
    to drop the rows of null values in genres and also in the column of original_title

In [2019]:
# dropping the column

df_movie_basics.drop(columns=['runtime_minutes'], inplace=True)

In [2020]:
#previewing the new dataset

df_movie_basics.head(3)

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   

   start_year              genres  
0        2013  Action,Crime,Drama  
1        2019     Biography,Drama  
2        2018               Drama

The runtime_minutes column has been drop. Now it is time to drop the Rows with null values in genres and the origina_title
It appropriate to replace the missing values in the rows of original_title because they are few missing values. It best to replace the missing values because the rows could be usefull in the analysis

In [2021]:
#dropping the rows with null values

df_movie_basics.dropna(inplace= True)

In [2022]:
# filling null values with a placeholder

df_movie_basics.fillna('MISSING', inplace=True)

In [2023]:
# checking for duplicates

df_movie_basics.duplicated(subset= ['primary_title', 'start_year']).sum()

1810

It appears that there are 1810 missing values in the dataset and it is best if we drop them

In [2024]:
# dropping the duplicate rows

df_movie_basics.drop_duplicates(subset= ['primary_title', 'start_year'], inplace = True)

Duplicated rows have been dropped 

In [2025]:
#preview the shape renewed datset

df_movie_basics.shape

(138924, 5)


The renewed dataset has 138924 Rows and 5 columns

In [2026]:
#getting info of the new dataset

df_movie_basics.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 138924 entries, 0 to 146143
Data columns (total 5 columns):
 #   Column          Non-Null Count   Dtype 
---  ------          --------------   ----- 
 0   movie_id        138924 non-null  object
 1   primary_title   138924 non-null  object
 2   original_title  138924 non-null  object
 3   start_year      138924 non-null  int64 
 4   genres          138924 non-null  object
dtypes: int64(1), object(4)
memory usage: 6.4+ MB


The dataset is now free of null values. It has 2 datatypes : interger and object 

In [2027]:
df_movie_basics.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year                genres  
0        2013    Action,Crime,Drama  
1        2019       Biography,Drama  
2        2018                 Drama  
3        2018          Comedy,Drama  
4        2017  Comedy,Drama,Fantasy

# 2.3.3 df_movie_ratings

In [2028]:
#preview of the dataset

df_movie_ratings.head()

movie_id  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [2029]:

# Accessing more information about df_movie_basics

df_movie_ratings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73856 entries, 0 to 73855
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   movie_id       73856 non-null  object 
 1   averagerating  73856 non-null  float64
 2   numvotes       73856 non-null  int64  
dtypes: float64(1), int64(1), object(1)
memory usage: 1.7+ MB


There are 3 data types in the dataset : object, float and interger and the data inside conforms to the data types.

It is time to check for missing values and duplicated values

In [2030]:
# checking for duplicate values

df_movie_ratings.duplicated(subset=['movie_id']).sum()

0

The dataset appears to not have duplicates

In [2031]:
#getting shape

df_movie_ratings.shape

(73856, 3)

The dataset has 73856 rows and 3 columns

In [2032]:
#checking for null values

df_movie_ratings.isna().sum()

movie_id         0
averagerating    0
numvotes         0
dtype: int64

The dataset 
has no null values

In [2033]:
#checking the descriptive statistics

df_movie_ratings.describe()

averagerating      numvotes
count   73856.000000  7.385600e+04
mean        6.332729  3.523662e+03
std         1.474978  3.029402e+04
min         1.000000  5.000000e+00
25%         5.500000  1.400000e+01
50%         6.500000  4.900000e+01
75%         7.400000  2.820000e+02
max        10.000000  1.841066e+06

Above are the descriptives of the dataset

In [2034]:
# check for duplicates in the data

df_movie_ratings.duplicated().sum()

0

The dataset has no null values

In [2035]:
#getting the shape of the dataset

df_movie_ratings.shape

(73856, 3)

# 2.4 Merging Datasets

In order to merge the data frames, a column is identified in the dataframes with the same data.
In the df_movie_basics and  df_movie_basics datasets, they all have a column with called movie_id and that what Will be appropriate 
to merge the the dataframes

In [2036]:
df_movie_info.dropna()

id                                           synopsis rating  \
1        3  New York City, not-too-distant-future: Eric Pa...      R   
6       10  Some cast and crew from NBC's highly acclaimed...  PG-13   
7       13  Stewart Kane, an Irishman living in the Austra...      R   
15      22  Two-time Academy Award Winner Kevin Spacey giv...      R   
18      25  From ancient Japan's most enduring tale, the e...  PG-13   
...    ...                                                ...    ...   
1530  1968  This holiday season, acclaimed filmmaker Camer...     PG   
1537  1976  Embrace of the Serpent features the encounter,...     NR   
1541  1980  A band of renegades on the run in outer space ...  PG-13   
1542  1981  Money, Fame and the Knowledge of English. In I...     NR   
1545  1985  A woman who joins the undead against her will ...      R   

                                                  genre             director  \
1                     Drama|Science Fiction and Fantasy     David Cronenberg   
6                                                Comedy          Jake Kasdan   
7                                                 Drama         Ray Lawrence   
15                    Comedy|Drama|Mystery and Suspense  George Hickenlooper   
18    Action and Adventure|Drama|Science Fiction and...     Carl Erik Rinsch   
...                                                 ...                  ...   
1530                                       Comedy|Drama        Cameron Crowe   
1537   Action and Adventure|Art House and International          Ciro Guerra   
1541   Action and Adventure|Science Fiction and Fantasy          Joss Whedon   
1542                                       Comedy|Drama         Gauri Shinde   
1545                        Horror|Mystery and Suspense  Sebastian Gutierrez   

                                    writer  theater_date      dvd_date  \
1             David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013   
6                               Mike White  Jan 11, 2002  Jun 18, 2002   
7         Raymond Carver|Beatrix Christian  Apr 27, 2006   Oct 2, 2007   
15                           Norman Snider  Dec 17, 2010   Apr 5, 2011   
18              Chris Morgan|Hossein Amini  Dec 25, 2013   Apr 1, 2014   
...                                    ...           ...           ...   
1530     Aline Brosh McKenna|Cameron Crowe  Dec 23, 2011   Apr 3, 2012   
1537  Ciro Guerra|Jacques Toulemonde Vidal  Feb 17, 2016  Jun 21, 2016   
1541                           Joss Whedon  Sep 30, 2005  Dec 20, 2005   
1542                          Gauri Shinde   Oct 5, 2012  Nov 20, 2012   
1545                   Sebastian Gutierrez   Jun 1, 2007   Oct 9, 2007   

     currency  box_office      runtime                  studio  
1           $     600,000  108 minutes       Entertainment One  
6           $  41,032,915   82 minutes      Paramount Pictures  
7           $     224,114  123 minutes  Sony Pictures Classics  
15          $   1,039,869  108 minutes            ATO Pictures  
18          $  20,518,224  127 minutes      Universal Pictures  
...       ...         ...          ...                     ...  
1530        $  72,700,000  126 minutes        20th Century Fox  
1537        $   1,320,005  123 minutes           Buffalo Films  
1541        $  25,335,935  119 minutes      Universal Pictures  
1542        $   1,416,189  129 minutes      Eros Entertainment  
1545        $      59,371   98 minutes        IDP Distribution  

[235 rows x 12 columns]

But first I should set the index of both dataframes to the to movie_id in order to be able to merge

In [2037]:
#setting movie id to be the index

df_movie_basics.set_index('movie_id')

primary_title  \
movie_id                                                 
tt0063540                                    Sunghursh   
tt0066787              One Day Before the Rainy Season   
tt0069049                   The Other Side of the Wind   
tt0069204                              Sabse Bada Sukh   
tt0100275                     The Wandering Soap Opera   
...                                                ...   
tt9916428                          The Secret of China   
tt9916538                          Kuambil Lagi Hatiku   
tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
tt9916706                              Dankyavar Danka   
tt9916754               Chico Albuquerque - Revelações   

                                        original_title  start_year  \
movie_id                                                             
tt0063540                                    Sunghursh        2013   
tt0066787                              Ashad Ka Ek Din        2019   
tt0069049                   The Other Side of the Wind        2018   
tt0069204                              Sabse Bada Sukh        2018   
tt0100275                        La Telenovela Errante        2017   
...                                                ...         ...   
tt9916428                          The Secret of China        2019   
tt9916538                          Kuambil Lagi Hatiku        2019   
tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
tt9916706                              Dankyavar Danka        2013   
tt9916754               Chico Albuquerque - Revelações        2013   

                          genres  
movie_id                          
tt0063540     Action,Crime,Drama  
tt0066787        Biography,Drama  
tt0069049                  Drama  
tt0069204           Comedy,Drama  
tt0100275   Comedy,Drama,Fantasy  
...                          ...  
tt9916428  Adventure,History,War  
tt9916538                  Drama  
tt9916622            Documentary  
tt9916706                 Comedy  
tt9916754            Documentary  

[138924 rows x 4 columns]

In [2038]:
# setting movie id to be the index

df_movie_ratings.set_index('movie_id')

averagerating  numvotes
movie_id                           
tt10356526            8.3        31
tt10384606            8.9       559
tt1042974             6.4        20
tt1043726             4.2     50352
tt1060240             6.5        21
...                   ...       ...
tt9805820             8.1        25
tt9844256             7.5        24
tt9851050             4.7        14
tt9886934             7.0         5
tt9894098             6.3       128

[73856 rows x 2 columns]

In [2039]:
#merging the 2 dataframes

df_basics_ratings = pd.merge(df_movie_basics, df_movie_ratings, how = 'inner')

df_basics_ratings.head(3)

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   

   start_year              genres  averagerating  numvotes  
0        2013  Action,Crime,Drama            7.0        77  
1        2019     Biography,Drama            7.2        43  
2        2018               Drama            6.9      4517

In [2040]:
df = pd.merge(df_movie_basics, df_movie_ratings, how = 'inner')

In [2041]:
df_basics_ratings_gross = pd.merge(df_basics_ratings, df_movie_gross , how = 'inner', 
                                   left_on = 'original_title', right_on = 'title')
df_basics_ratings_gross

movie_id                    primary_title  \
0     tt0337692                      On the Road   
1     tt4339118                      On the Road   
2     tt5647250                      On the Road   
3     tt0359950  The Secret Life of Walter Mitty   
4     tt0365907      A Walk Among the Tombstones   
...         ...                              ...   
1514  tt7690670                         SuperFly   
1515  tt7784604                       Hereditary   
1516  tt8011712                         The Past   
1517  tt8097306                    Nobody's Fool   
1518  tt9151704        Burn the Stage: The Movie   

                       original_title  start_year                   genres  \
0                         On the Road        2012  Adventure,Drama,Romance   
1                         On the Road        2014                    Drama   
2                         On the Road        2016                    Drama   
3     The Secret Life of Walter Mitty        2013   Adventure,Comedy,Drama   
4         A Walk Among the Tombstones        2014       Action,Crime,Drama   
...                               ...         ...                      ...   
1514                         Superfly        2018    Action,Crime,Thriller   
1515                       Hereditary        2018     Drama,Horror,Mystery   
1516                         The Past        2018             Drama,Horror   
1517                    Nobody's Fool        2018     Comedy,Drama,Romance   
1518        Burn the Stage: The Movie        2018        Documentary,Music   

      averagerating  numvotes                            title     studio  \
0               6.1     37886                      On the Road        IFC   
1               6.0         6                      On the Road        IFC   
2               5.7       127                      On the Road        IFC   
3               7.3    275300  The Secret Life of Walter Mitty        Fox   
4               6.5    105116      A Walk Among the Tombstones       Uni.   
...             ...       ...                              ...        ...   
1514            5.0      4753                         Superfly       Sony   
1515            7.3    151571                       Hereditary        A24   
1516            7.4        54                         The Past        SPC   
1517            4.6      3618                    Nobody's Fool       Par.   
1518            8.8      2067        Burn the Stage: The Movie  Trafalgar   

      domestic_gross foreign_gross  year  
0             744000       8000000  2012  
1             744000       8000000  2012  
2             744000       8000000  2012  
3           58200000     129900000  2013  
4           26300000      26900000  2014  
...              ...           ...   ...  
1514        20500000        236000  2018  
1515        44100000      35300000  2018  
1516         1300000       9300000  2013  
1517        31700000       1800000  2018  
1518         4200000      16100000  2018  

[1519 rows x 12 columns]

The dataframe has 12 columns.The columns of primary_title, secondary_title and title all represent the same thing.
Just like start_year and year whic represent the same value
It is appropriate that I drop all unneccessary columns

I am going to drop the primary_title, original_title and year

In [2042]:
# dropping unnecessary columns

df_basics_ratings_gross.drop(columns=['primary_title', 'original_title', 'year'], inplace=True)

In [2043]:
#previewing the datset 

df_basics_ratings_gross.head()

movie_id  start_year                   genres  averagerating  numvotes  \
0  tt0337692        2012  Adventure,Drama,Romance            6.1     37886   
1  tt4339118        2014                    Drama            6.0         6   
2  tt5647250        2016                    Drama            5.7       127   
3  tt0359950        2013   Adventure,Comedy,Drama            7.3    275300   
4  tt0365907        2014       Action,Crime,Drama            6.5    105116   

                             title studio  domestic_gross foreign_gross  
0                      On the Road    IFC          744000       8000000  
1                      On the Road    IFC          744000       8000000  
2                      On the Road    IFC          744000       8000000  
3  The Secret Life of Walter Mitty    Fox        58200000     129900000  
4      A Walk Among the Tombstones   Uni.        26300000      26900000

In [2044]:
df_movie_info

id                                           synopsis rating  \
0        1  This gritty, fast-paced, and innovative police...      R   
1        3  New York City, not-too-distant-future: Eric Pa...      R   
2        5  Illeana Douglas delivers a superb performance ...      R   
3        6  Michael Douglas runs afoul of a treacherous su...      R   
4        7                                                NaN     NR   
...    ...                                                ...    ...   
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   
1556  1997  The popular Saturday Night Live sketch was exp...     PG   
1557  1998  Based on a novel by Richard Powell, when the l...      G   
1558  1999  The Sandlot is a coming-of-age story about a g...     PG   
1559  2000  Suspended from the force, Paris cop Hubert is ...      R   

                                                  genre            director  \
0                   Action and Adventure|Classics|Drama    William Friedkin   
1                     Drama|Science Fiction and Fantasy    David Cronenberg   
2                     Drama|Musical and Performing Arts      Allison Anders   
3                            Drama|Mystery and Suspense      Barry Levinson   
4                                         Drama|Romance      Rodney Bennett   
...                                                 ...                 ...   
1555   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1556                 Comedy|Science Fiction and Fantasy        Steve Barron   
1557  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1558    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
1559  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
0                                       Ernest Tidyman   Oct 9, 1971   
1                         David Cronenberg|Don DeLillo  Aug 17, 2012   
2                                       Allison Anders  Sep 13, 1996   
3                      Paul Attanasio|Michael Crichton   Dec 9, 1994   
4                                         Giles Cooper           NaN   
...                                                ...           ...   
1555                                               NaN  Aug 18, 2006   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1557                                               NaN   Jan 1, 1962   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
1559                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office      runtime             studio  
0     Sep 25, 2001      NaN         NaN  104 minutes                NaN  
1      Jan 1, 2013        $     600,000  108 minutes  Entertainment One  
2     Apr 18, 2000      NaN         NaN  116 minutes                NaN  
3     Aug 27, 1997      NaN         NaN  128 minutes                NaN  
4              NaN      NaN         NaN  200 minutes                NaN  
...            ...      ...         ...          ...                ...  
1555   Jan 2, 2007        $  33,886,034  106 minutes    New Line Cinema  
1556  Apr 17, 2001      NaN         NaN   88 minutes  Paramount Vantage  
1557  May 11, 2004      NaN         NaN  111 minutes                NaN  
1558  Jan 29, 2002      NaN         NaN  101 minutes                NaN  
1559  Feb 11, 2003      NaN         NaN   94 minutes  Columbia Pictures  

[1560 rows x 12 columns]

The dataset has 2 data types : interger and float.

In [2045]:
# checking for null values

df_basics_ratings_gross.isna().sum()

movie_id          0
start_year        0
genres            0
averagerating     0
numvotes          0
title             0
studio            0
domestic_gross    0
foreign_gross     0
dtype: int64

The final combined dataset has no nullvales

In [2046]:
# checking for duplicates

df_basics_ratings_gross.duplicated(['movie_id']).sum()

0

The combined datasets has no dupicates

In [2047]:
#new order columns

column_names = 'movie_id', 'title', 'genres','studio', 'start_year', 'numvotes', 'averagerating', 'domestic_gross', 'averagerating', 'foreign_gross'

In [2048]:
# reordering the columns

df_basics_ratings_gross.reindex(columns=column_names)

movie_id                            title                   genres  \
0     tt0337692                      On the Road  Adventure,Drama,Romance   
1     tt4339118                      On the Road                    Drama   
2     tt5647250                      On the Road                    Drama   
3     tt0359950  The Secret Life of Walter Mitty   Adventure,Comedy,Drama   
4     tt0365907      A Walk Among the Tombstones       Action,Crime,Drama   
...         ...                              ...                      ...   
1514  tt7690670                         Superfly    Action,Crime,Thriller   
1515  tt7784604                       Hereditary     Drama,Horror,Mystery   
1516  tt8011712                         The Past             Drama,Horror   
1517  tt8097306                    Nobody's Fool     Comedy,Drama,Romance   
1518  tt9151704        Burn the Stage: The Movie        Documentary,Music   

         studio  start_year  numvotes  averagerating  domestic_gross  \
0           IFC        2012     37886            6.1          744000   
1           IFC        2014         6            6.0          744000   
2           IFC        2016       127            5.7          744000   
3           Fox        2013    275300            7.3        58200000   
4          Uni.        2014    105116            6.5        26300000   
...         ...         ...       ...            ...             ...   
1514       Sony        2018      4753            5.0        20500000   
1515        A24        2018    151571            7.3        44100000   
1516        SPC        2018        54            7.4         1300000   
1517       Par.        2018      3618            4.6        31700000   
1518  Trafalgar        2018      2067            8.8         4200000   

      averagerating foreign_gross  
0               6.1       8000000  
1               6.0       8000000  
2               5.7       8000000  
3               7.3     129900000  
4               6.5      26900000  
...             ...           ...  
1514            5.0        236000  
1515            7.3      35300000  
1516            7.4       9300000  
1517            4.6       1800000  
1518            8.8      16100000  

[1519 rows x 10 columns]

In [2049]:
df_basics_ratings_gross

movie_id  start_year                   genres  averagerating  numvotes  \
0     tt0337692        2012  Adventure,Drama,Romance            6.1     37886   
1     tt4339118        2014                    Drama            6.0         6   
2     tt5647250        2016                    Drama            5.7       127   
3     tt0359950        2013   Adventure,Comedy,Drama            7.3    275300   
4     tt0365907        2014       Action,Crime,Drama            6.5    105116   
...         ...         ...                      ...            ...       ...   
1514  tt7690670        2018    Action,Crime,Thriller            5.0      4753   
1515  tt7784604        2018     Drama,Horror,Mystery            7.3    151571   
1516  tt8011712        2018             Drama,Horror            7.4        54   
1517  tt8097306        2018     Comedy,Drama,Romance            4.6      3618   
1518  tt9151704        2018        Documentary,Music            8.8      2067   

                                title     studio  domestic_gross foreign_gross  
0                         On the Road        IFC          744000       8000000  
1                         On the Road        IFC          744000       8000000  
2                         On the Road        IFC          744000       8000000  
3     The Secret Life of Walter Mitty        Fox        58200000     129900000  
4         A Walk Among the Tombstones       Uni.        26300000      26900000  
...                               ...        ...             ...           ...  
1514                         Superfly       Sony        20500000        236000  
1515                       Hereditary        A24        44100000      35300000  
1516                         The Past        SPC         1300000       9300000  
1517                    Nobody's Fool       Par.        31700000       1800000  
1518        Burn the Stage: The Movie  Trafalgar         4200000      16100000  

[1519 rows x 9 columns]

In [2050]:
# Exporting this combined dataframe to a csv

df_basics_ratings_gross.to_csv('Combined Microsoft Data.csv')

# 3. Exploratory Data Analysis (EDA)¶